src/main/java/appeng/integration/modules/CraftGuide.java

```diff
@@ -100,11 +92,22 @@ public void generateRecipes( RecipeGenerator generator )
 
 		RecipeTemplate shapelessTemplate = new DefaultRecipeTemplate( this.shapelessCraftingSlots, RecipeGeneratorImplementation.workbench, new TextureClip( DynamicTexture.instance( "recipe_backgrounds" ), 1, 121, 79, 58 ), new TextureClip( DynamicTexture.instance( "recipe_backgrounds" ), 82, 121, 79, 58 ) );
 
-		RecipeTemplate furnaceTemplate = new DefaultRecipeTemplate( this.furnaceSlots, new ItemStack( Blocks.furnace ), new TextureClip( DynamicTexture.instance( "recipe_backgrounds" ), 1, 181, 79, 58 ), new TextureClip( DynamicTexture.instance( "recipe_backgrounds" ), 82, 181, 79, 58 ) );
+		this.addCraftingRecipes( craftingTemplate, smallTemplate, shapelessTemplate, generator );
+
+		final IAppEngApi api = AEApi.instance();
+		IBlocks aeBlocks = api.definitions().blocks();
```

Comment:
We usually mark variable/fields as `final` to prevent unintentional mutation of them.
As this is an ongoing project, not every class has already be changed. 
It would be nice, if you could change it and save us a bit of work.